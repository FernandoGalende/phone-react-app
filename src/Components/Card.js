import React, { Component } from 'react';
import PropTypes from 'prop-types'
import { Link } from 'react-router-dom'

export class Card extends Component {
  static propTypes = {
    id: PropTypes.string,
    title: PropTypes.string,
    color: PropTypes.string,
    image: PropTypes.string
  }

  render(){
    const { id, title, image, color } = this.props
    return (
      <Link to={`/detail/${id}`} className="card">
        <div className="card-image">
          <figure className="image">
            <img
              src={image}
              alt={title}
            />
          </figure>
        </div>
        <div className="card-content">
          <div className="media">
            <div className="media-content">
              <p className="title is-4">{ title }</p>
              <p className="subtitle is-6">{ color }</p>
            </div>
          </div>
        </div>
      </Link>
    )
  }
}
