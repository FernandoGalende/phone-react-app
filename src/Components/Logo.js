import React, { Component } from 'react';
import logo from '../assets/logo.svg';

class Logo extends Component {
  render() {
    return (
      <div>
        <img
          src={logo}
          className="App-logo"
          alt="logo"
        />
    </div>
    )
  }
}

export default Logo
