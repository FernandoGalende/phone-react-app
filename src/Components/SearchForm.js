import React, { Component } from 'react';

const BASE_URL=process.env.REACT_APP_BASEURL

export class SearchForm extends Component {
  options = {
    method: 'GET',
    headers: { 'Access-Control-Allow-Origin': '*'}
  }

  _handleChange = (e) => {
    this.setState({inputMovie: e.target.value})
  }

  _handleSubmit = (e) => {
    e.preventDefault()

    fetch(`${BASE_URL}/phones`,this.options)
    .then( res => res.json() )
    .then( results => {
       this.props.onResults( results.data )
    })
  }

  render () {
    return(
      <form onSubmit={ this._handleSubmit } id="search-form">
        <div className="field has-addons">
          <div className="control">
            <input
              className="input"
              onChange={ this._handleChange }
              type="text"
              placeholder="Find a phone"/>
          </div>
          <div className="control">
            <button id="search-button"
              className="button is-info">
              Search
            </button>
          </div>
        </div>
      </form>
    )
  }
}
