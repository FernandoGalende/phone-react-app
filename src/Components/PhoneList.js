import React, { Component } from 'react';
import PropTypes from 'prop-types'

import { Card } from '../Components/Card'

export class PhoneList extends Component {
  static propTypes = {
    phones: PropTypes.array,
  }

  render(){
    const { phones } = this.props
    return(
      <div className="PhonesList">

      {
        phones.map(phone => {
          return (
            <div
              key={ phone.id }
              className="PhonesList-Item">
                <Card
                  id = { phone.id }
                  title = { phone.title }
                  color = { phone.color }
                  image = { phone.thumbnails }
                />
            </div>
          )
        })
      }
      </div>
    )
  }
}