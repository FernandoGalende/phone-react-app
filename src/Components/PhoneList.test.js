import React from 'react';
import { shallow } from 'enzyme'

//Components
import { PhoneList } from '../Components/PhoneList'

//assets
import mockResults from '../assets/mockDataTest'


describe('PhoneList component', () => {
  it('should render PhoneList component with properly elements', () => {
    const PhoneListComponent = shallow(<PhoneList phones = { mockResults.data } />
    )
    expect(PhoneListComponent).toMatchSnapshot()
  });
})
