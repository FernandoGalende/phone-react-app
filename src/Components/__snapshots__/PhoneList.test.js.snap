// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneList component should render PhoneList component with properly elements 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PhoneList
    phones={
      Array [
        Object {
          "color": "Gold",
          "description": "Cupertino, California — Apple today announced iPhone Xs and iPhone Xs Max, the most advanced iPhones ever, taking the vision for the future of the smartphone to a new level. The 5.8-inch iPhone Xs and 6.5-inch iPhone Xs Max feature stunning Super Retina displays, a faster and improved dual camera system that offers breakthrough photo and video features, the first 7-nanometer chip in a smartphone — the A12 Bionic chip with next-generation Neural Engine — faster Face ID, wider stereo sound, a beautiful new gold finish and introduce Dual SIM to iPhone. iPhone Xs and iPhone Xs Max will be available for pre-order beginning Friday, September 14 and in stores beginning Friday, September 21.",
          "id": "01",
          "image": "https://cdn.tmobile.com/content/dam/t-mobile/en-p/cell-phones/apple/apple-iphone-xs/gold/Apple-iPhoneXs-Gold-1-3x.jpg",
          "specs": Object {
            "dimensions": Object {
              "size": "5.94 x 2.98 x 0.33",
            },
            "display": Object {
              "liquidRetinaHDDisplay": "The iPhone XR display has rounded corners that follow a beautiful curved design, and these corners are within a standard rectangle. When measured as a standard rectangular shape, the screen is 6.06 inches diagonally (actual viewable area is less).",
              "resolution": "1792-by-828-pixel",
            },
          },
          "thumbnails": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
          "title": "iPhone XS Max",
        },
        Object {
          "color": "Black",
          "description": "Cupertino, California — Apple today announced iPhone Xs and iPhone Xs Max, the most advanced iPhones ever, taking the vision for the future of the smartphone to a new level. The 5.8-inch iPhone Xs and 6.5-inch iPhone Xs Max feature stunning Super Retina displays, a faster and improved dual camera system that offers breakthrough photo and video features, the first 7-nanometer chip in a smartphone — the A12 Bionic chip with next-generation Neural Engine — faster Face ID, wider stereo sound, a beautiful new gold finish and introduce Dual SIM to iPhone. iPhone Xs and iPhone Xs Max will be available for pre-order beginning Friday, September 14 and in stores beginning Friday, September 21.",
          "id": "02",
          "image": "https://cdn.tmobile.com/content/dam/t-mobile/en-p/cell-phones/apple/apple-iphone-xr/black/Apple-iPhoneXr-Black-1-3x.jpg",
          "specs": Object {
            "dimensions": Object {
              "size": "5.94 x 2.98 x 0.33",
            },
            "display": Object {
              "liquidRetinaHDDisplay": "The iPhone XR display has rounded corners that follow a beautiful curved design, and these corners are within a standard rectangle. When measured as a standard rectangular shape, the screen is 6.06 inches diagonally (actual viewable area is less).",
              "resolution": "1792-by-828-pixel",
            },
          },
          "thumbnails": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
          "title": "iPhone XR",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="PhonesList-Item"
        >
          <Card
            color="Gold"
            id="01"
            image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            title="iPhone XS Max"
          />
        </div>,
        <div
          className="PhonesList-Item"
        >
          <Card
            color="Black"
            id="02"
            image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            title="iPhone XR"
          />
        </div>,
      ],
      "className": "PhonesList",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "01",
        "nodeType": "host",
        "props": Object {
          "children": <Card
            color="Gold"
            id="01"
            image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            title="iPhone XS Max"
          />,
          "className": "PhonesList-Item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "Gold",
            "id": "01",
            "image": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
            "title": "iPhone XS Max",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": "02",
        "nodeType": "host",
        "props": Object {
          "children": <Card
            color="Black"
            id="02"
            image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
            title="iPhone XR"
          />,
          "className": "PhonesList-Item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "color": "Black",
            "id": "02",
            "image": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
            "title": "iPhone XR",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="PhonesList-Item"
          >
            <Card
              color="Gold"
              id="01"
              image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              title="iPhone XS Max"
            />
          </div>,
          <div
            className="PhonesList-Item"
          >
            <Card
              color="Black"
              id="02"
              image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              title="iPhone XR"
            />
          </div>,
        ],
        "className": "PhonesList",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "01",
          "nodeType": "host",
          "props": Object {
            "children": <Card
              color="Gold"
              id="01"
              image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              title="iPhone XS Max"
            />,
            "className": "PhonesList-Item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "Gold",
              "id": "01",
              "image": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
              "title": "iPhone XS Max",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "02",
          "nodeType": "host",
          "props": Object {
            "children": <Card
              color="Black"
              id="02"
              image="https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png"
              title="iPhone XR"
            />,
            "className": "PhonesList-Item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "color": "Black",
              "id": "02",
              "image": "https://s.tmocache.com//images/png/products/phones/Apple-iPhone-XS-Gold/170x250.png",
              "title": "iPhone XR",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
